### DISCUSSION TITLE
Release for Mac?

### DISCUSSION THREAD

**Discussion by rtrahan504** [intel/openvino-plugins-ai-audacity#274] [General]
Is there anything inherently in the source code or dependencies that is incompatible to Mac? I don't see anything with a quick spot check. I'm also not a Mac owner or user so I can't even attempt. A friend is hoping for a MacOS release.

**Comment by RyanMetcalfeInt8** [comment]
Hi @rtrahan504,

> Is there anything inherently in the source code or dependencies that is incompatible to Mac?

Nope, not that I can think of. Building for Mac would probably be more similar to building for Linux, which I have instructions posted here: https://github.com/intel/openvino-plugins-ai-audacity/blob/main/doc/build_doc/linux/README.md

But I haven't tried, as I don't have a Mac either :)

Happy to review any changes needed to support Mac if someone from the community wants to take a shot.

Thanks,
Ryan

**Reply by GigiCarlo68** [reply]
Its over a year old .. is there one for mac by now ? Instructions ?

**Reply by RyanMetcalfeInt8** [reply]
Not much has changed in a year. Our team still doesn't have Macs, nor the bandwidth to develop for them.

**Comment by rtrahan504** [comment]
Thanks @RyanMetcalfeInt8. Everything seems like standard C++. I've honestly never used a Mac, so I don't know where to start. The linux build seems standard too except for a lot of manual building of packages. Maybe just leave this issue up and a Mac user might chime in?

**Comment by RyanMetcalfeInt8** [comment]
> The linux build seems standard too except for a lot of manual building of packages.

Yeah, a couple packages need to be built manually -- they're not too bad though.

> Maybe just leave this issue up and a Mac user might chime in? 

Sure, let me tag with 'Good first issue' -- maybe someone will volunteer. Cheers!

**Comment by Jackmattson** [comment]
I'm not a developer per say but quite surprised no one has jumped on this yet.

**Comment by kingrollo** [comment]
I wasn't going to respond until I'd finished this, but since you asked...

I don't recall ever building anything on MacOS before, so this is a bit of an adventure, but I've been working on it and am making some progress. There are some tricks and tips that I've worked through and will try to document properly, but this is roughly where I've got to:

*Dependencies*
`brew install openvino opencv libtorch cmake`

I think you can then skip everything (although uncertain if libtorch still needs its path exporting) until:

*Whisper.cpp*
This section all works the same way as at [the link](https://github.com/intel/openvino-plugins-ai-audacity/blob/main/doc/build_doc/linux/README.md) except you need to change `cmake ../whisper.cpp/ -DWHISPER_OPENVINO=ON` to `cmake ../ -DWHISPER_OPENVINO=ON`

*OpenVINO Stable-Diffusion CPP*
Works as per Linux docs.

*Audacity initial (vanilla) build*
This is a bit different to the Linux docs, as libasound2-dev seems to be Linux-only and I'm not certain I've translated all the other Homebrew dependencies correctly:
```
#Install some build dependencies
brew install git python3-pip
sudo pip3 install conan
brew install gtk+ 
brew install jack
brew install ossp-uuid
# clone Audacity
git clone https://github.com/audacity/audacity.git
# Check out Audacity-3.4.2 tag, 
cd audacity
git checkout Audacity-3.4.2
cd ..
# Create build directory
mkdir audacity-build
cd audacity-build
# Run cmake (grab a coffee & a snack... this takes a while)
cmake -G "Unix Makefiles" ../audacity -DCMAKE_BUILD_TYPE=Release
# build it 
make -j`nproc`
```
That then builds a broken app, which is probably to be expected given the hamfisted approach I've been taking, but it's getting late...

**Comment by gblong1** [comment]
Hey Roland - What generation mac are you using?

**Comment by kingrollo** [comment]
MacOS 12.7 Monterey on a 2016 MacBook Pro with Intel.

**Comment by kingrollo** [comment]
I've poked around a bit more but am a bit stuck at this point, and unsure what direction to take next. The Mac build instructions for Audacity are a bit sparse. The [main instructions](https://github.com/audacity/audacity/blob/master/BUILDING.md#macos-1) describe using the XCode IDE, which might work - I don't actually know it, but I can't imagine it's insurmountable. Would the next Linux step (the OpenVINO module) work if you built it through the IDE? I'd test it, but XCode requires a more recent version of MacOS.

The [signing instructions](https://github.com/audacity/audacity/blob/master/mac/signing.txt) refer to a file called cmake_build.txt, but that was deleted [three years ago](https://github.com/audacity/audacity/commit/66a85bf97c70149b2da61c892148bd1139ab1533) and doesn't [appear to be](https://github.com/audacity/audacity/blob/b4d6595ff9f1729460a25b0c0ff3f5c62cc857df/mac/cmake_build.txt) fully current / relevant.

**Comment by RyanMetcalfeInt8** [comment]
> Would the next Linux step (the OpenVINO module) work if you built it through the IDE? 

Yes, it should. You should be able to drop the ```mod-openvino``` folder in ```src/modules/```, modify ```src/modules/CMakeLists.txt``` and then go through the IDE build steps as usual.

> The [signing instructions](https://github.com/audacity/audacity/blob/master/mac/signing.txt) refer to a file called cmake_build.txt, but that was deleted [three years ago](https://github.com/audacity/audacity/commit/66a85bf97c70149b2da61c892148bd1139ab1533) and doesn't [appear to be](https://github.com/audacity/audacity/blob/b4d6595ff9f1729460a25b0c0ff3f5c62cc857df/mac/cmake_build.txt) fully current / relevant.

So Mac requires you to sign the binary to use them or something?

**Comment by ashleycox** [comment]
I've been looking into this too as I'd love to have this working on the Mac, but haven't been able to get further than the discussion here and no luck building audacity via the IDE> Anyone else made progress?

**Comment by cshaver** [comment]
I was able to build vanilla Audacity, and then with the `mod-openvino.so` in the modules folder, but when I restart Audacity with the module enabled I get an error loading the module:
```
Unable to load the module "/path/to/audacity-build/Release/Audacity.app/Contents/modules/mod-openvino.so". Error: No such file or directory
```

---

the main things I did differently than the instructions y'all mentioned were using the homebrew-installed libtorch
```
# Libtorch
export LIBTORCH_ROOTDIR=/opt/homebrew/include/torch/
```

and I got an error about an ambiguous usage in `mod-openvino/htdemucs.h` so I tweaked this bit :hammer: 

```
#include <torch/torch.h>
// namespace torch
// {
//     class Tensor;
// }
```

**Comment by RyanMetcalfeInt8** [comment]
> I was able to build vanilla Audacity, and then with the `mod-openvino.so` in the modules folder, but when I restart Audacity with the module enabled I get an error loading the module:
> 
> ```
> Unable to load the module "/path/to/audacity-build/Release/Audacity.app/Contents/modules/mod-openvino.so". Error: No such file or directory
> ```
 hmm, I wonder if it is having trouble resolving some other .so's? I'm not a Mac developer so maybe this tip wouldn't apply, but from the terminal that you run Audacity from, could you do something like ```ldd /path/to/audacity-build/Release/Audacity.app/Contents/modules/mod-openvino.so``` and see if it lists any dependencies as ```Not Found```?

**Comment by cshaver** [comment]
> hmm, I wonder if it is having trouble resolving some other .so's? I'm not a Mac developer so maybe this tip wouldn't apply, but from the terminal that you run Audacity from, could you do something like `ldd /path/to/audacity-build/Release/Audacity.app/Contents/modules/mod-openvino.so` and see if it lists any dependencies as `Not Found`?

Google led me to `otool`, which it [sounds like](https://stackoverflow.com/questions/45464584/macosx-which-dynamic-libraries-linked-by-binary#:~:text=ldd%20actually%20looks%20for%20the,see%20if%20they%20are%20there.) doesn't actually look at the libraries the way `ldd` does.

<details><summary><code>otool -L Debug/Audacity.app/Contents/modules/mod-openvino.so</code></summary>
<pre>
@loader_path/../Frameworks/libopenvino.2330.dylib (compatibility version 2330.0.0, current version 2023.3.0)
@loader_path/../Frameworks/libc10.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwhisper.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libstable_diffusion_ov.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libstable_diffusion_audio_ov.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-theme-resources.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-graphics.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-project-file-io.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-wx-init.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-time-track.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-snapping.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-audio-io.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-import-export.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-tags.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-export-ui.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-shuttlegui.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-theme.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-wx-wrappers.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_osx_cocoau_xrc-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_osx_cocoau_qa-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_osx_cocoau_aui-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_osx_cocoau_html-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_baseu_xml-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libexpat.1.8.10.dylib (compatibility version 10.0.0, current version 10.10.0)
@loader_path/../Frameworks/libwx_baseu_net-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_osx_cocoau_adv-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_osx_cocoau_core-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libpng16d.16.dylib (compatibility version 16.0.0, current version 16.39.0)
@loader_path/../Frameworks/libturbojpeg.0.dylib (compatibility version 0.0.0, current version 0.2.0)
@loader_path/../Frameworks/libjpeg.8.dylib (compatibility version 8.0.0, current version 8.2.2)
@loader_path/../Frameworks/lib-file-formats.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-vst.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-lv2.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-ladspa.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-vst3.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-audio-unit.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-effects.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-wave-track.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-stretching-sequence.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-time-and-pitch.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-sample-track.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-mixer.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-audio-graph.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-playable-track.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-track-selection.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-track.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-time-frequency-selection.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-screen-geometry.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-command-parameters.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-numeric-formats.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-project-rate.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-audio-devices.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-realtime-effects.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-project-history.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-project.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-transactions.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-math.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-channel.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-registries.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-module-manager.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-xml.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-string-utils.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-files.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-preferences.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-components.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-exceptions.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-basic-ui.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-strings.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libwx_baseu-3.1.dylib (compatibility version 0.0.0, current version 0.0.0)
/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 24.0.0)
/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
/System/Library/Frameworks/AVKit.framework/Versions/A/AVKit (compatibility version 1.0.0, current version 1.0.0)
/System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation (compatibility version 1.0.0, current version 2.0.0)
/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 2202.0.0)
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
/System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices (compatibility version 1.0.0, current version 64.0.0)
/System/Library/Frameworks/CoreText.framework/Versions/A/CoreText (compatibility version 1.0.0, current version 1.0.0)
/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1774.2.3)
/System/Library/Frameworks/CoreMedia.framework/Versions/A/CoreMedia (compatibility version 1.0.0, current version 1.0.0)
/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 61040.61.1)
/System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO (compatibility version 1.0.0, current version 1.0.0)
/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1336.61.1)
@loader_path/../Frameworks/lib-ipc.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/lib-utility.dylib (compatibility version 0.0.0, current version 0.0.0)
/System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit (compatibility version 1.0.0, current version 1.0.0)
@loader_path/../Frameworks/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
@loader_path/../Frameworks/libportmidi.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libportmidi_s.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libsndfile.1.dylib (compatibility version 1.0.0, current version 1.0.31)
@loader_path/../Frameworks/libvorbisenc.2.0.12.dylib (compatibility version 2.0.12, current version 0.0.0)
@loader_path/../Frameworks/libvorbis.0.4.9.dylib (compatibility version 0.4.9, current version 0.0.0)
@loader_path/../Frameworks/libFLAC++.10.dylib (compatibility version 10.0.0, current version 10.0.0)
@loader_path/../Frameworks/libFLAC.12.dylib (compatibility version 12.0.0, current version 12.0.0)
@loader_path/../Frameworks/libogg.0.dylib (compatibility version 0.0.0, current version 0.8.5)
@loader_path/../Frameworks/libopus.0.dylib (compatibility version 0.0.0, current version 0.9.0)
@loader_path/../Frameworks/libportaudio.dylib (compatibility version 0.0.0, current version 0.0.0)
/System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio (compatibility version 1.0.0, current version 1.0.0)
/System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox (compatibility version 1.0.0, current version 1000.0.0)
/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1226.0.0)
/System/Library/Frameworks/Carbon.framework/Versions/A/Carbon (compatibility version 2.0.0, current version 170.0.0)
@loader_path/../Frameworks/libtorch.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libtorch_cpu.dylib (compatibility version 0.0.0, current version 0.0.0)
@loader_path/../Frameworks/libprotobuf.25.3.0.dylib (compatibility version 0.0.0, current version 25.3.0)
@loader_path/../Frameworks/libabsl_log_internal_check_op.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_leak_check.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_die_if_null.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_conditions.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_message.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_nullguard.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_examine_stack.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_format.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_proto.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_log_sink_set.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_sink.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_entry.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_marshalling.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_reflection.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_config.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_program_name.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_private_handle_accessor.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_commandlineflag.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_flags_commandlineflag_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_initialize.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_globals.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_internal_globals.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_raw_hash_set.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_hash.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_city.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_low_level_hash.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_hashtablez_sampler.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_statusor.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_status.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_cord.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_cordz_info.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_cord_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_cordz_functions.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_exponential_biased.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_cordz_handle.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_crc_cord_state.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_crc32c.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_crc_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_crc_cpu_detect.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_bad_optional_access.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_str_format_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_strerror.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_synchronization.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_stacktrace.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_symbolize.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_debugging_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_demangle_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_graphcycles_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_kernel_timeout_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_malloc_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_time.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_strings.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_string_view.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_throw_delegate.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_strings_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_base.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_spinlock_wait.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_int128.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_civil_time.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_time_zone.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 2202.0.0)
@loader_path/../Frameworks/libabsl_bad_variant_access.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_raw_logging_internal.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
@loader_path/../Frameworks/libabsl_log_severity.2308.0.0.dylib (compatibility version 2308.0.0, current version 0.0.0)
/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1600.157.0)
</pre>
</details>

I only looked at the couple of absolute paths like `/usr/lib/libc++.1.dylib` and didn't see the file there on my system, however using `otool -L` on the `mod-ffmpeg.so` uses it too so it can't be that strange. I'll keep poking around!

**Comment by eukarpov** [comment]
@cshaver It looks like you were almost getting a working build.

It can work on Apple M1. 

![image](https://github.com/intel/openvino-plugins-ai-audacity/assets/1791423/2c2fdb69-4f96-431e-be92-1beeba7bcf9d)

To resolve references issues and make models available:

1. prevent apple security check for openvino_toolkit libraries if you trust the source
use  "xattr -d com.apple.quarantine openvino_toolkit_*/runtime/lib/arm64/Release/*.so " and for dylib
2. copy openvino_toolkit_*/runtime/lib/arm64/Release/*.so and *.dylib to audacity-build/Release/Audacity.app/Contents/Frameworks
3. copy libtbb.12.dylib to audacity-build/Release/3rdparty/tbb/libtbb.12.dylib
4. copy models to audacity-build/Release/Audacity.app/Contents/openvino-models

**Comment by flohma** [comment]
This would be so cool to use on a Mac with M1 chip. Has anyone built a working plugin for that yet?

**Comment by eukarpov** [comment]
Yes, the plug-in works for me on Apple M1. I am preparing a build on GitHub.

**Comment by flohma** [comment]
That would be great @eukarpov for the likes of me who're not good with Terminal...

**Comment by PaperTheGoose** [comment]
@eukarpov will the build work for user that don’t have M1?

**Comment by eukarpov** [comment]
The build will work for Apple M generation devices as it is based on Arm64. For x64 devices it should be another build.

**Comment by Jackmattson** [comment]
Has anyone posted an installer yet?

**Comment by shock** [comment]
@eukarpov 
I see you started a pull request yesterday.  It looks like you aborted it?  I don't have a lot of time, but I do have a working Xcode 15 build environment on an M1 Mac.  Let me know if you need any help testing anything.  I will try to reproduce the steps I _think_ you are taking locally as soon as I have time.  If you can add any details or list out your full steps in here, that could make it more clear what you've done.

**Comment by eukarpov** [comment]
@shock 
The work on creating a Mac build on GitHub was started 2 days ago, and a new PR had to be created to test a build workflow. It was mistakenly made directly to this repo. The PR has been closed, and it will be reopened once it works in my fork.

There was progress on obtaining build artifacts; however, it still lacks some dependencies for the installation to be completed. This weekend could be a date estimation for when the artifacts might be deployed correctly.

Please join review 
https://github.com/eukarpov/openvino-plugins-ai-audacity/pull/1
The latest build artificats
https://github.com/eukarpov/openvino-plugins-ai-audacity/actions/runs/8561925816?pr=1

Thanks!

**Comment by shock** [comment]
@eukarpov: Great!  Sometime today, I'll fork your fork, and have a look at your GitHub build work.  I'll message you in your repo...

**Comment by eukarpov** [comment]
[How to install openvino-plugins-ai-audacity on macOS Arm64](https://github.com/eukarpov/openvino-plugins-ai-audacity/issues/3)

**Comment by danruzicka** [comment]
Any information or  status "HOw to install openvino plugin aduacity on Mac OS Intel"?

**Comment by eukarpov** [comment]
> Any information or status "HOw to install openvino plugin aduacity on Mac OS Intel"?

I will allocate some time next week to investigate possibility of an x64 build on macOS.

**Comment by danruzicka** [comment]
> > Any information or status "HOw to install openvino plugin aduacity on Mac OS Intel"?
> 
> I will allocate some time next week to investigate possibility of an x64 build on macOS.

I’m looking forward to it.

**Comment by danruzicka** [comment]
@eukarpov how's your investigation going? Will that be possible or not? Thanks for the answer!

**Comment by eukarpov** [comment]
@danruzicka Sorry, I was busy with other things these days. I will try to compile it next week.

**Comment by ra7kar** [comment]
@eukarpov, Thank you so much for your efforts in working on the build for a Mac.
I tried https://github.com/eukarpov/openvino-plugins-ai-audacity/issues/3, but could not make it to work. I have a Mac M1 Pro, and MacOS Sonoma 14.4.1, also have Audacity 3.5.1 install on my laptop. 
I will attempt again once you release the steps and the new build. Thank you

**Comment by 7ratcat7** [comment]
Hi! Any update on an installer for mac?

**Comment by doctorhy** [comment]
I have a Macbook Pro (Apple M3 Pro); how to install the plugins for Audacity? Thanks

**Comment by Openstreetmapler** [comment]
While there’s no OpenVINO installer for Mac yet and if you specifically want the **music separation** feature, you can download the latest version of **[DemucsGUI](https://github.com/CarlGao4/Demucs-Gui/releases)**. It allows you to choose the model (default _htdemucs_ should be good for most cases) and tweak more settings than in the OpenVINO interface. (Please choose an output filepath you want, the default may not be good!) You can drag-and-drop files and press the start button to separate the stems.
